sound/soc/sof/loader.c

```diff
@@ -128,6 +128,7 @@ int snd_sof_parse_module_memcpy(struct snd_sof_dev *sdev,
 		case SOF_FW_BLK_TYPE_IRAM:
 		case SOF_FW_BLK_TYPE_DRAM:
 			offset = block->offset;
+			bar = snd_sof_dsp_get_bar_index(sdev, block->type);
```

Comment:
@dbaluta According to your definition of snd_sof_dsp_get_bar_index(), it can return a negative error code. I think you want to check for it. Also, make the "bar" variable just "int" to account for errors.

Reply:
@lyakh good point. Fixed and pushed the changes.